File: Intro_POO/Intro_POO_python/Persona.py
class Persona:
    def __init__(self, nombre, paterno, materno, edad, ci):
        self.nombre = nombre
        self.paterno = paterno
        self.materno = materno
        self.edad = edad
        self.ci = ci

    def mostrarDatos(self):
        print(f"Nombre: {self.nombre} {self.paterno} {self.materno}")
        print(f"Edad: {self.edad}")
        print(f"CI: {self.ci}")
        print("-------------------------")

    def mayorDeEdad(self):
        return self.edad >= 18

    def modificarEdad(self, nuevo):
        self.edad = nuevo


    def getPaterno(self):
        return self.paterno

if __name__ == "__main__":

    p1 = Persona("Carlos", "Perez", "Lopez", 20, "1234567")
    p2 = Persona("Ana", "Perez", "Gomez", 17, "9876543")

    p1.mostrarDatos()
    p2.mostrarDatos()

    print("Carlos es mayor de edad" if p1.mayorDeEdad() else "Carlos es menor de edad")
    print("Ana es mayor de edad" if p2.mayorDeEdad() else "Ana es menor de edad")

    p2.modificarEdad(18)
    print("Nueva edad de Ana: ")
    p2.mostrarDatos()

    if p1.getPaterno() == p2.getPaterno():
        print(f"Ambos tienen el mismo apellido paterno: {p1.getPaterno()}")
    else:
        print("Tienen diferentes apellidos paternos.")

File: Intro_POO/Ejercicio_5/Persona.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package com.mycompany.introduccionpoo;

/**
 *
 * @author HP-RYZEN 5
 */
//Nicol Yanina Ticona Aduviri
//R.U: 1855987
class Persona {
    private String nombre;
    private String paterno;
    private String materno;
    private int edad;
    private String ci;

    // Constructor
    public Persona(String nombre, String paterno, String materno, int edad, String ci) {
        this.nombre = nombre;
        this.paterno = paterno;
        this.materno = materno;
        this.edad = edad;
        this.ci = ci;
    }

    // Mostrar datos
    public void mostrarDatos() {
        System.out.println("Nombre: " + nombre + " " + paterno + " " + materno);
        System.out.println("Edad: " + edad);
        System.out.println("CI: " + ci);
        System.out.println("-------------------------");
    }

    // Verificar si es mayor de edad
    public boolean mayorDeEdad() {
        return edad >= 18;
    }

    // Modificar edad
    public void modificarEdad(int nuevo) {
        this.edad = nuevo;
    }

    // Getter del apellido paterno
    public String getPaterno() {
        return paterno;
    }
}
